hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/JettyHttpMessageSenderProvider.java

```diff
@@ -51,21 +63,40 @@ public MessageSender create(Subscription subscription) {
 
     private HttpRequestFactory httpRequestFactory(Subscription subscription) {
         int requestTimeout = subscription.getSerialSubscriptionPolicy().getRequestTimeout();
-        return new HttpRequestFactory(httpClient, requestTimeout, metadataAppender, authorizationProviderFactory.create(subscription));
+        HttpClient client = subscription.isHttp2Enabled() ? http2Client.orElseGet(() -> {
+            logger.info("Http2 delivery is not enabled on this server. Falling back to http1.");
+            return httpClient;
+        }) : httpClient;
+        return new HttpRequestFactory(http2Client.get(), requestTimeout, metadataAppender, authorizationProviderFactory.create(subscription));
```

Comment:
We need to remember about switching the http client for given subscription when the subscription is being updated (the `http2Enabled` property). This requires change in `ConsumerMessageSender.updateSubscription(subscription)`.